src/App.tsx

```diff
@@ -1,24 +1,49 @@
-import React, { useState, useCallback } from 'react';
+import React, { useState, useCallback, useEffect } from 'react';
 import axios from 'axios';
 import styled from 'styled-components';
 import PhotosTable from './PhotosTable';
 import PhotoMetaData from './PhotoMetaData';
 
 import './App.css';
-import { Photo } from './types';
+import { Photo, Facets, GraphState, GraphType, GRAPH_TYPES } from './types';
 import Navbar from './layout/Navbar';
 import FileUplaoder from './FileUploader';
+import GraphContainer from './GraphContainer';
 
 const App: React.FC = () => {
   const [photos, updatePhotos] = useState<Photo[]>([]);
   const [currentPhoto, updateCurrentPhoto] = useState<Photo | null>(null);
+  const [currentFacets, updateFacets] = useState<Facets>({});
+  const [leftGraphState, updateLeftGraph] = useState<GraphState | null>(null);
+  const [rightGraphState, updateRightGraph] = useState<GraphState | null>(null)
 
   const handleSearchSubmit = useCallback((searchQuery: string) => {
-    axios.post('http://localhost:8080/api/v1/photos', { text_query: searchQuery }).then((response) => {
+    axios.post('http://localhost:8080/api/v1/photos', { text_query: searchQuery, facets: GRAPH_TYPES }).then((response) => {
       updatePhotos(response.data.photos);
+      updateFacets(response.data.facets);
     });
   }, []);
 
+  useEffect( () => {
+    async function fetchInitalData() {
+      await axios.post('http://localhost:8080/api/v1/photos', { facets: GRAPH_TYPES }).then((response) => {
+        let facets = response.data.facets
+
+        updatePhotos(response.data.photos);
```

Comment:
so instead of updating state 4 times in this callback you can create single action type for `useReducer` like: `InitialDataFetchSuccess` and set everything there. 
This is also non performant since updating state in async callback cannot be batched by React and will result in 4 consecutive renders of component.